### DISCUSSION TITLE
grpc客户端的生效范围是什么，用一个节点的ip打开dashboard配置grpc客户端， 是只有这个节点生效还是集群所有节点都会启动grpc客户端？

### DISCUSSION THREAD

**Discussion by shadowycgg** [emqx/emqx#12358] [Q&A]
如题。
另外，假设只有一个节点运行客户端， 那么这个客户端是会上报集群所有的mqtt钩子事件，还是只有与这个节点建立tcp长链接的mqtt客户端的事件？
能否有方法设置或者配置是在单节点还是在集群中所有节点启动grpc客户端，而不是一个个去启动（假如情况是第一种的话）

**Comment by savonarola** [comment]
Hello!

If you mean "Parser gRPC" resourse, then each node will connect.

**Reply by shadowycgg** [reply]
Thanks for your answer,
I mean grpc exhook (in EMQX document, a exhook is a grpc server so I need to start a grpc client in emqx dashboard) in other language, so I  want  to know if I configure a  exhook in emqx dashboard, is that all cluster node will start a grpc client for same server address or just the the node that dashboard I opened in.
but I don't know what "parser gRPC" is, is that the same thing？

**Comment by shadowycgg** [comment]
经过验证，集群中每个节点上的钩子函数都会单独执行，但是貌似grpc客户端只能在集群内全局生效，我的目的是希望每个节点都有grpc客户端的配置，将钩子事件发送到不同的grpc server上，这个有办法做到吗？
